src/main/java/com/kmush12/CurrencyConverter/Converter/CurrencyConverterService.java

```diff
@@ -34,14 +37,17 @@ public ExchangeResult createExchangeResult(ExchangeRequest exchangeRequest) thro
                 .orElseThrow(() -> new Exception("Exchange rate for a given currency set cannot be found"));
     }
 
-    public Optional<ExchangeRate> findExchangeRate(ExchangeRequest request) {
-        return exchangeRatesList.stream()
-                .filter(exchangeRate -> exchangeRate.originCurrency().equals(request.originCurrency()) && exchangeRate.destinationCurrency().equals(request.destinationCurrency()))
-                .findFirst();
-
+    private Optional<ExchangeRate> getExchangeRateFromAPI(ExchangeRequest exchangeRequest, String pathToConversionRate) throws IOException {
```

Comment:
Osobna klasa plus to mapowanie obiektów można zrobić trochę ładniej z użyciem jednego z frameworków, które podałem wyżej.